test/common/webdriver_bidi.py

```diff
@@ -292,6 +293,16 @@ def switch_to_top(self) -> None:
         self.context = self.top_context
         log_command.info("← switch_to_top")
 
+    @contextlib.contextmanager
+    def restore_context(self, *, switch_to_top: bool = True) -> Iterator[None]:
+        saved = self.context
+        if switch_to_top:
```

Comment:
It was a surprise to find this here. I think it would be cleaner to not change context in `restore_context`.

Reply:
@mvollmer You are right. Changed in #20907